### PR TITLE
fix(gpu): Fix GPU Arm-2D

### PR DESCRIPTION
**Pull request by GorgonMeducer**
Following the new Driver APIs

### Description of the feature or fix

Fixed the GPU Arm-2D

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: src/draw/arm2d/lv_gpu_arm2d.c [hunk 1]
```diff
@@ -84,12 +87,12 @@
...
     __arm_2d_impl_rgb565_colour_filling_mask_opacity
 #define __arm_2d_impl_copy              __arm_2d_impl_rgb16_copy
-#define __arm_2d_impl_alpha_blending    __arm_2d_impl_rgb565_alpha_blending
+#define __arm_2d_impl_tile_copy_opacity    __arm_2d_impl_rgb565_tile_copy_opacity
 #define __arm_2d_impl_src_msk_copy      __arm_2d_impl_rgb565_src_msk_copy
 #define __arm_2d_impl_src_chn_msk_copy  __arm_2d_impl_rgb565_src_chn_msk_copy
 #define __arm_2d_impl_cl_key_copy       __arm_2d_impl_rgb16_cl_key_copy
-#define __arm_2d_impl_alpha_blending_colour_keying                              \
-    __arm_2d_impl_rgb565_alpha_blending_colour_keying
+#define __arm_2d_impl_tile_copy_colour_keying_opacity                           \
```
Comment (kisvegabor) [inline src/draw/arm2d/lv_gpu_arm2d.c]: FYI, on other places in LVGL we use the US "color" spelling. As these defines are not part of the public API we can stick with yours/ARMs flavor.

**Review by kisvegabor — COMMENTED** [review]
Thank you! I had only a minor comment. Feel free to ignore it and let me know if I can merge this PR.

**Comment by GorgonMeducer** [comment]
I have replaced `colour` with `color` when possible. Please review them.
If possible, please merge this PR. 

Thank you.

**Comment by kisvegabor** [comment]
> I have replaced colour with color when possible. Please review them.

I'm not sure mixing colour and color is better than using colour only. As this driver is maintained by you and it doesn't affect other parts  I leave it to you. Feel free to pick the option you prefer the most. 

---

I think you'd like to merge the fix of #4029 too, so I wait for it.

**Comment by GorgonMeducer** [comment]
Also about #4030

**Comment by GorgonMeducer** [comment]
@kisvegabor done

**Comment by GorgonMeducer** [comment]
@kisvegabor You can merge it now.

**Pull request merged**